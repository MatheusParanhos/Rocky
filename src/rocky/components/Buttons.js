import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Scissors from "./ScissorsButton";
import Paper from "./PaperButton";
import Rock from "./RockButton";
export default class Buttons extends Component {
  render() {
    return (
      <View style={styles.buttonsContainer}>
        
        <View style={styles.bottom}>
          <Rock onPress={this.props.onRockPress}/>
        </View>
        <View style={styles.top}>
          <Paper onPress={this.props.onPaperPress}/>
          <Scissors onPress={this.props.onScissorsPress}/>
        </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  buttonsContainer: {
    height: 300,
    width: '100%',
   //  backgroundColor: colors.gray,

   // borderRadius: 25,
   flexDirection:'column'
  },
  top: {
      flex:1,
    height: 100,
    width: deviceWidth / 2,
    flexDirection: "row",
    // backgroundColor: "yellow"
  },
  bottom: {
      flex:1,
    height: 100,
    width: deviceWidth,
   // backgroundColor: "orange",
    alignItems:'center',
    justifyContent:'center',
  }
});
