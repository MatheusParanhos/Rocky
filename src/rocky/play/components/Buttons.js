import React, { Component } from "react";
import { Text, View, StyleSheet } from "react-native";
import { deviceWidth } from "../../../theme/theme";

import PropTypes from "prop-types";
import Scissors from "./ScissorsButton";
import Paper from "./PaperButton";
import Rock from "./RockButton";
export default class Buttons extends Component {
  render() {
    return (
      <View style={styles.buttonsContainer}>
        {/* <View style={styles.bottom}>
          
        </View> */}
        <View style={styles.bottom}>
          <Rock onPress={this.props.onRockPress} />
          <Paper onPress={this.props.onPaperPress} />
          <Scissors onPress={this.props.onScissorsPress} />
        </View>
      </View>
    );
  }
}
Buttons.propTypes = {
  onRockPress: PropTypes.func.isRequired,
  onPaperPress: PropTypes.func.isRequired,
  onScissorsPress: PropTypes.func.isRequired,
  userChoice: PropTypes.string,
  computerChoice: PropTypes.string,
  timer: PropTypes.number
};
Buttons.defaultProps = {
  timer: 0,
  result: "",
  userChoice: "",
  computerChoice: "",
  onRockPress: () => {},
  onPaperPress: () => {},
  onScissorsPress: () => {}
};
const styles = StyleSheet.create({
  buttonsContainer: {
    height: 300,
    width: "100%",
    //  backgroundColor: colors.gray,

    // borderRadius: 25,
    flexDirection: "column"
  },
  bottom: {
    flex: 1,
    height: 100,
    width: deviceWidth,
    flexDirection: "row",
    // backgroundColor: "orange",
    alignItems: "center",
    justifyContent: "space-around"
  }
});
