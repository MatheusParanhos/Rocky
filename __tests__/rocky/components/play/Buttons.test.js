import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Buttons from '../../../../src/rocky/play/components/Buttons';
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe('Testing Buttons component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Buttons onScissorsPress={()=>console.log('scissors')} onPaperPress={()=>console.log('paper')} onRockPress={()=>console.log('rock')} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});