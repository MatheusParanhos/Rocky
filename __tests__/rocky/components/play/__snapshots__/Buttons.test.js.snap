// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Buttons component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    computerChoice=""
    onPaperPress={[Function]}
    onRockPress={[Function]}
    onScissorsPress={[Function]}
    result=""
    timer={0}
    userChoice=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "space-around",
            "width": 750,
          }
        }
      >
        <Rock
          onPress={[Function]}
        />
        <Paper
          onPress={[Function]}
        />
        <Scissors
          onPress={[Function]}
        />
      </Component>,
      "style": Object {
        "flexDirection": "column",
        "height": 300,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Rock
            onPress={[Function]}
          />,
          <Paper
            onPress={[Function]}
          />,
          <Scissors
            onPress={[Function]}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "space-around",
          "width": 750,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-around",
              "width": 750,
            }
          }
        >
          <Rock
            onPress={[Function]}
          />
          <Paper
            onPress={[Function]}
          />
          <Scissors
            onPress={[Function]}
          />
        </Component>,
        "style": Object {
          "flexDirection": "column",
          "height": 300,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Rock
              onPress={[Function]}
            />,
            <Paper
              onPress={[Function]}
            />,
            <Scissors
              onPress={[Function]}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "space-around",
            "width": 750,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Testing Buttons component renders as expected 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    computerChoice=""
    count={1}
    onPaperPress={[Function]}
    onRockPress={[Function]}
    onScissorsPress={[Function]}
    result=""
    timer={0}
    userChoice=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "space-around",
            "width": 750,
          }
        }
      >
        <Rock
          onPress={[Function]}
        />
        <Paper
          onPress={[Function]}
        />
        <Scissors
          onPress={[Function]}
        />
      </Component>,
      "style": Object {
        "flexDirection": "column",
        "height": 300,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Rock
            onPress={[Function]}
          />,
          <Paper
            onPress={[Function]}
          />,
          <Scissors
            onPress={[Function]}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "space-around",
          "width": 750,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-around",
              "width": 750,
            }
          }
        >
          <Rock
            onPress={[Function]}
          />
          <Paper
            onPress={[Function]}
          />
          <Scissors
            onPress={[Function]}
          />
        </Component>,
        "style": Object {
          "flexDirection": "column",
          "height": 300,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Rock
              onPress={[Function]}
            />,
            <Paper
              onPress={[Function]}
            />,
            <Scissors
              onPress={[Function]}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "space-around",
            "width": 750,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
